docs/Technical.md

```diff
@@ -0,0 +1,138 @@
+The solution will be deployed thanks to Kubernetes service on Azure.
+
+### **Front-end**
+
+On the website, a search bar will be displayed in which the user has to enter the adress he wants to know when Santa will arrive. Then, if the adress typed is correct, a countdown until Santa arrives at the midnight of the sun at this adress will be displayed.
+To calculate when Santa will arrive, we will add or subtract time depending on if the adress if at a negative or postive according to GMT (greenwich meridian). The countdown will be displayed in an DD/HH/MM/SS.
+A Reindeer wiil be displayed on the bottom left of the page. It will tell the user what to type in the searchbar, or warn the user if what they typed in is in a valid format (For more info go see the [error messages](#3c-error-messages)).
+In order to ease the use of the website by users, we will have cookies in which we will store the adress typed, that we will be activited only if the user agrees to it. Thanks to this, the user will not have to type his adress each time is going on the website.
+The website will be accessible through any web browser. 
+
+### **Back-end**
+
+The website will be deployed on an nginx server. The nginx server will take care of redirecting the adress type to the database, process the calculations to get the remaining time until Christmas and sending back the info to the front-end.
+
+### **Database**
+
+For our database, we will use the data from [geofabrik](http://download.geofabrik.de). This dataset is quite heavy but it allows us to have a great precision, and it has also regular updates. The data will be stored in a PostgreSQL database.
+In the case where this database is not fast enough or if is manipulation is too complicated, we could use data from this [dataset](http://www.geonames.org). This dataset is less precise because it only take in account the city and not the complete adress, but it is less heavier and easier to manipulate than the data from openstreetmap.
+The database will store:
+
+- The adress
+- The postal code
+- The longitudes of a place
+- The country of the postal adress
+
+## 3. Further Considerations
+
+### **3.a Cost analysis**
+
+- We will use the 200€ free credits provided by Azure, so the project should cost 0€.
+
+### **3.b Security**
+
+### Security threats
+
+- Someone could get the address or the GPS coordinates of the users.
+
+- A lot of people could try to connect at the same time at the service.
+
+### Security measures
+
+- The data that a user will type on the website will not be registered anywhere to be sure that nobody can access a user's location.
+
+- Thanks to kubernetes service from Azure, the website will auto-scale to adapt to the inflow.
+
+### Privacy considerations
+
+- Only the adress typed could be stored in cookies, only if the user wants to
+
+### Risks
+
+- The connection between azure and the website could crash, leading to an incapacity to access the website.
+
```

Comment:
What if we run out of credits on Azure ?

Reply:
done